tests/test_serializer_nested.py

```diff
@@ -69,3 +69,59 @@ def test_multipart_validate(self):
         input_data = QueryDict('nested[one]=1')
         serializer = self.Serializer(data=input_data)
         assert serializer.is_valid()
+
+
+class TestNestedSerializerWithMany:
+    def setup(self):
+        class NestedSerializer(serializers.Serializer):
+            example = serializers.IntegerField(max_value=10)
+
+        class TestSerializer(serializers.Serializer):
+            allow_null = NestedSerializer(many=True, allow_null=True)
+            not_allow_null = NestedSerializer(many=True)
+
+        self.Serializer = TestSerializer
+
+    def test_null_allowed_if_allow_null_is_set(self):
+        input_data = {
+            'allow_null': None,
+            'not_allow_null': [{'example': '2'}, {'example': '3'}]
+        }
+        expected_data = {
+            'allow_null': None,
+            'not_allow_null': [{'example': 2}, {'example': 3}]
+        }
+        serializer = self.Serializer(data=input_data)
+
+        assert serializer.is_valid(), serializer.errors
+        assert serializer.validated_data == expected_data
+
+    def test_null_is_not_allowed_if_allow_null_is_not_set(self):
+        input_data = {
+            'allow_null': None,
+            'not_allow_null': None
+        }
+        serializer = self.Serializer(data=input_data)
+
+        assert not serializer.is_valid()
+
+        expected_errors = {'not_allow_null': [serializer.error_messages['null']]}
+        assert serializer.errors == expected_errors
+
+    def test_run_the_field_validation_even_if_the_field_is_null(self):
```

Comment:
I understand the other two cases, but the motivation of this one is unclear to me.
Could you run me through an example of why this behavior is desired?
Alternatively, we could simply drop this test?